### PR TITLE
Add state and searchbar with filter functionality

### PR DESCRIPTION
**Pull request by DTThor**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,11 +1,37 @@
...
 import Movie from './components/Movie'
 import MovieList from './components/MovieList'
+import SearchBar from './components/SearchBar'
+import movies from './movies.js'
+
+class App extends Component {
+  constructor() {
+    super()
+    this.state = {
+      movies: movies,
```
Comment (lucas-aragno) [inline src/App.js]: what would movies be here?

File: src/App.js [hunk 2]
```diff
@@ -1,11 +1,37 @@
...
+    this.onSearchChange = this.onSearchChange.bind(this)
+  }
+
+  onSearchChange(e){
+    this.setState({ searchfield: e.target.value })
+  }
+
+  render(){
+    const { movies, searchfield } = this.state;
+    const filteredMovies = movies.filter(movie => {
```
Comment (lucas-aragno) [inline src/App.js]: move this outside the render method

File: src/App.js [hunk 3]
```diff
@@ -1,11 +1,41 @@
...
 import MovieList from './components/MovieList'
+import SearchBar from './components/SearchBar'
+import movies from './movies.js'
+
+class App extends Component {
+  constructor() {
+    super()
+    this.state = {
+      movies,
+      searchfield: ''
```
Comment (lucas-aragno) [inline src/App.js]: use camelCase for `searchfield`

Comment (DTThor) [reply src/App.js]: done! Thanks.

**Pull request merged**